File: ios-native/Packages/Sources/DataModule/Networking/APIConfiguration.swift
import Foundation

/// API 連線設定：base URL 與授權碼。由 App target（composition root）注入，
/// 讓 base URL / 金鑰不散落在 DataModule 內。
public struct APIConfiguration: Sendable {
    public let baseURL: URL
    public let authorizationKey: String

    public init(baseURL: URL, authorizationKey: String) {
        self.baseURL = baseURL
        self.authorizationKey = authorizationKey
    }
}

File: flutter/lib/core/network/api_configuration.dart
/// API 連線設定：base URL 與授權碼。由 composition root（router）注入，
/// 讓 base URL / 金鑰不散落在資料層。與 iOS 端 `APIConfiguration` 對齊。
class ApiConfiguration {
  const ApiConfiguration({required this.baseUrl, required this.authorizationKey});

  final String baseUrl;
  final String authorizationKey;
}
